src/app/DeviceSelector.tsx

```diff
@@ -59,9 +62,10 @@ const deviceListing: DeviceTraits = {
  * Note that the callbacks releaseCurrentDevice and onDeviceIsReady
  * are only invoked, if a deviceSetup is defined.
  */
-const onDeviceSelected = (dispatch: AppDispatch) => (device: Device) => {
+const onDeviceSelected = (dispatch: AppDispatch) => async (device: Device) => {
     logger.info(`Selected device with s/n ${device.serialNumber}`);
-    getBoardControllerVersion(dispatch, device); // FIXME: Remove this when onDeviceIsReady() is called
+    await getBoardControllerVersion(dispatch, device); // FIXME: Remove this when onDeviceIsReady() is called
+    await getCurrentBoardControllerConfig(dispatch, device); // FIXME: Remove this when onDeviceIsReady() is called
```

Comment:
Guess this isn't important, I just don't understand the reason for why not.
But you can also wrap this within dispatch and change the definition for the functions to be
const x = (y) => dispatch => {}

Reply:
I have looked into this briefly earlier with Ådne, although I didn't change it. I'll look into it later on.